app/models/data_file.rb

```diff
@@ -15,7 +15,7 @@ class DataFile < ApplicationRecord
   # allow same titles, but only if these belong to different users
   # validates_uniqueness_of :title, :scope => [ :contributor_id, :contributor_type ], :message => "error - you already have a Data file with such title."
 
-  has_one :content_blob, ->(r) { where('content_blobs.asset_version =?', r.version) }, as: :asset, foreign_key: :asset_id
+  has_one :content_blob, ->(r) { where('content_blobs.asset_version =? AND deleted =?', r.version, false) }, as: :asset, foreign_key: :asset_id
```

Comment:
I assume this works for existing content blobs with a `NULL` value in the `deleted` column

Reply:
the migration sets the default to false, which I've always assumed updates previous records. I've checked my local database and this does seem to be the case.